File: ios/dodamdodamappWatch Watch App/Feature/Timetable/TimetableModel.swift
import Foundation

struct TimetablePeriod: Identifiable {
  let id = UUID()
  let period: Int
  let subject: String
}

enum TimetableCardState {
  case loaded(dayLabel: String, periods: [TimetablePeriod], currentPeriod: Int?)
  case weekend
  case empty(dayLabel: String)
}

File: android/wear/src/main/java/com/b1nd/dodam/student/watch/feature/timetable/TimetableModel.kt
package com.b1nd.dodam.student.watch.feature.timetable

data class TimetablePeriod(val period: Int, val subject: String)

sealed class TimetableCardState {
  data class Loaded(val dayLabel: String, val periods: List<TimetablePeriod>, val currentPeriod: Int?) : TimetableCardState()
  data object Weekend : TimetableCardState()
  data class Empty(val dayLabel: String) : TimetableCardState()
}
